torchtext/datasets/ag_news.py

```diff
@@ -30,11 +31,18 @@
 
 @_add_docstring_header(num_lines=NUM_LINES, num_classes=4)
 @_create_dataset_directory(dataset_name=DATASET_NAME)
-@_wrap_split_argument(('train', 'test'))
-def AG_NEWS(root, split):
-    path = download_from_url(URL[split], root=root,
-                             path=os.path.join(root, split + ".csv"),
-                             hash_value=MD5[split],
-                             hash_type='md5')
-    return _RawTextIterableDataset(DATASET_NAME, NUM_LINES[split],
-                                   _create_data_from_csv(path))
+@_wrap_split_argument(("train", "test"))
+def AG_NEWS(root: str, split: Union[Tuple[str], str]):
+    if not is_module_available("torchdata"):
+        raise ModuleNotFoundError("Package `torchdata` not found. Please install following instructions at `https://github.com/pytorch/data`")
+
+    url_dp = IterableWrapper([URL[split]])
+    http_dp = HttpReader(url_dp)
+    cache_dp = http_dp.on_disk_cache(
+        filepath_fn=lambda x: os.path.join(root, split + ".csv"),
+        hash_dict={os.path.join(root, split + ".csv"): MD5[split]},
+        hash_type="md5"
+    ).end_caching(mode="w", same_filepath_fn=True)
```

Comment:
nit: can we update this to follow the same sequence of datapipes url->cache->download->end_caching as oppose to url->download->cache->end_caching as demonstrated in example [here](https://github.com/pytorch/data/blob/ec32ee42983760640280c937868611355e6ed281/torchdata/datapipes/iter/util/cacheholder.py#L130) and done for AmazonReviewPolarity/Full datasets? Not sure if there are any (performance) advantage by following one pattern over another. @ejguan could you please share any insights here?

Reply:
Yes and you should do so. The following pattern would re-download the file that violates the hash.
```suggestion
    cache_dp = url_dp.on_disk_cache(
        filepath_fn=lambda x: os.path.join(root, split + ".csv"),  # Here may need to change this function
        hash_dict={os.path.join(root, split + ".csv"): MD5[split]},
        hash_type="md5"
    )
    cache_dp = HttpReader(cache_dp)
    cache_dp = cache_dp.end_caching(mode="w", same_filepath_fn=True)
```

Reply:
Sounds good -- I'll make the fix in a moment!